soil-moisture-monitor.ino

```diff
@@ -117,3 +143,25 @@ void flashScreen() {
   }
 }
 
+/*
+ * Return current temperature in Celcius or -1.0 in case of communication error.
+ */
+float getTempInCelcius() {
+  float temp = dht.readTemperature();
+  if (!isnan(temp))
+    return temp;
```

Comment:
We can assume that the temp will be always positive and convert any potential negative value here to zero. Just in case of getting confused by reading -1*C from the sensor.